tests/app/notifications/rest/test_send_notification.py

```diff
@@ -24,21 +23,34 @@
     sample_api_key as create_sample_api_key
 )
 
+from app.models import Template
+from app.errors import InvalidRequest
 
-def test_create_sms_should_reject_if_missing_required_fields(notify_api, sample_api_key, mocker):
+
+@pytest.mark.parametrize('template_type',
+                         ['sms', 'email'])
+def test_create_notification_should_reject_if_missing_required_fields(notify_api,
+                                                                      sample_api_key, mocker, template_type):
     with notify_api.test_request_context():
         with notify_api.test_client() as client:
-            mocker.patch('app.celery.provider_tasks.deliver_sms.apply_async')
+            if template_type == 'sms':
```

Comment:
Rather than have an `if` statement here you could pass in the thing to be patched as part of the `parametrize`, eg:
```
@pytest.mark.parametrize('template_type, patch_target, task', [
    (
        'sms',
        'app.celery.provider_tasks.deliver_sms.apply_async',
        app.celery.provider_tasks.deliver_sms.apply_async
    ),
    …
])
```